rewrite-maven/src/main/java/org/openrewrite/maven/UpgradeDependencyVersion.java

```diff
@@ -402,6 +407,37 @@ private String resolveVersion(String version) {
                 return MavenDependency.findNewerVersion(groupId, artifactId, version, getResolutionResult(), metadataFailures,
                         versionComparator, ctx);
             }
+
+            /**
+             * Gets the requested group id for the provided artifact id from the pom
+             *
+             * @return The groupd if for the requested artifact id or null
+             */
+            @Nullable
+            private String getRequestedGroupIdForArtifact(String artifactId) {
+                return getResolutionResult().getPom().getRequested().getDependencyManagement().stream()
```

Comment:
Probably best not to use streams here as this recipe is potentially ran quite a few times in a single run and the overhead of the intermediate objects could be too high